tools/src/wyvern/target/corewyvernIL/decl/TypeDeclaration.java

```diff
@@ -36,7 +39,7 @@ public void setSourceType(Type sourceType) {
 
 	@Override
 	public DeclType typeCheck(TypeContext ctx, TypeContext thisCtx) {
-		// TODO Auto-generated method stub
-		return null;
+		DeclType declt = new ValDeclType(super.getName(), (ValueType) this.sourceType); 
```

Comment:
This should return an AbstractTypeMember or ConcreteTypeMember, depending on whether sourceType is specified or not.  It should not return a ValDeclType, as that is the type of a val declaration.